deploy/crds/apps_v1alpha1_servicebindingrequest_crd.yaml

```diff
@@ -69,6 +69,22 @@ spec:
               - kind
               - resourceRef
               type: object
+            customEnvVar:
```

Comment:
Would much rather see the ```SecretTransforms``` structure from ServiceBinding/AppBinding (https://github.com/kmodules/custom-resources/blob/114abf10dfe25d845bf8a96c8e5820ee2427aab4/apis/appcatalog/v1alpha1/appbinding_types.go#L47) used here rather than this new ```CustomEnvVar``` structure whose name seems to restrict its use for environment variables only and not volume mounts.

Reply:
Right, I'll let this PR move in for now - and then have the custom volume mounts addressed separately. We'll make a breaking API change if needed. Your point is valid ( CustomEnvVar is too specific ..) . https://github.com/redhat-developer/service-binding-operator/issues/172

@ron1 , In our case, `SecretTransforms` is not entirely valid because I'm trying to see if the binding can be done merely by the user expressing what she needs by looking at the backing service CR , even if there's nothing in the OLM descriptor. ( we have a long term goal to make this friendly for non-OLM users as well, not sure if we'll accomplish that though ) :)

Reply:
OLM Annotations + Secret Transforms is a well-understood tech combo that could work today and solve most known use cases. However, it does require buy-in from the OLM backing service operator community and exposes JSON Path which is not entirely user-friendly.

OTOH, I find the current use of both OLM Annotations AND the ```customEnvVar``` struct confusing. In particular, it is not clear that ```customEnvVar``` is both an alternative to OLM Annotations for locating interesting keys to be included in the binding secret and a mechanism to transform binding secret values. It is also not clear whether the transform language references the names of OLM annotation fields or paths to backing service CR fields.

Reply:
> In particular, it is not clear that customEnvVar is both an alternative to OLM

It intends to be, but it isn't yet :)